src/orientationUtils/Utils.java

```diff
@@ -1,112 +1,115 @@
-	 * @param side2
-	 *            another side of the triangle
-	 * @param angle
-	 *            the angle between the two sides above
-	 * @return the third side of the triangle, calculated by the formula: s3 =
-	 *         sqrt(s1^2+s2^2-2*s1*s2*cos(angle))
-	 */
-	public static double findThirdSide(double side1, double side2, double angle) {
-		return Math.sqrt(side1 * side1 + side2 * side2 - 2 * side1 * side2 * Math.cos(angle));
-	}
-
+package orientationUtils;
+
+import org.apache.commons.math3.linear.Array2DRowRealMatrix;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.apache.commons.math3.linear.RealVector;
+
+import utils.Point;
+
+public class Utils {
+
+	public static RealMatrix getTransformationMatrix(double yaw, double roll, double pitch) {
+		RealMatrix result = new Array2DRowRealMatrix(3, 3);
+
+		// defining variables for the sine and cosine results of the angles
+		double cosRoll = Math.cos(roll);
+		double cosYaw = Math.cos(yaw);
+		double cosPitch = Math.cos(pitch);
+
+		double sinRoll = Math.sin(roll);
+		double sinYaw = Math.sin(yaw);
+		double sinPitch = Math.sin(pitch);
+
+		// calculating each value of the matrix
+		result.setEntry(0, 0, cosYaw * cosPitch);
+		result.setEntry(0, 1, cosYaw * sinPitch * sinRoll - cosYaw * sinRoll);
+		result.setEntry(0, 2, cosYaw * sinPitch * cosRoll + sinYaw * sinRoll);
+
+		result.setEntry(1, 0, sinYaw * cosPitch);
+		result.setEntry(1, 1, sinYaw * sinPitch * sinRoll + cosYaw * cosRoll);
+		result.setEntry(1, 2, sinYaw * sinPitch * cosRoll - cosYaw * sinRoll);
+
+		result.setEntry(2, 0, -sinPitch);
+		result.setEntry(2, 1, cosPitch * sinRoll);
+		result.setEntry(2, 2, cosPitch * cosRoll);
+
+		return result;
+	}
+
+	public static RealMatrix getUnitMatrix(int size) {
+		RealMatrix result = new Array2DRowRealMatrix(size, size);
+
+		for (int i = 0; i < result.getRowDimension(); i++)
+			result.setEntry(i, i, 1);
+
+		return result;
+	}
+
+	/**
+	 * multiplies a given vector by itself, element by element
+	 * 
+	 * @param vector
+	 *            the vector which should be squared
+	 * @return a vector, containing the squares of the elements in the given
+	 *         vector
+	 */
+	public static RealVector getSquared(RealVector vector) {
+		return vector.ebeMultiply(vector);
+	}
+
+	/**
+	 * calculates and returns the two legs of a triangle by its hypotenuse and
+	 * the argument facing the Y leg.
+	 * 
+	 * @param hypotenuse
+	 *            the hypotenuse's length
+	 * @param argument
+	 *            the argument facing the Y leg.
+	 * @return a point containing the lengths of the X and Y legs of the
+	 *         triangle in that order: (X,Y)
+	 */
+	public static Point findLegsByHypotenuseAndArg(double hypotenuse, double argument) {
+		// calculating the lengths of the X and Y legs
+		double legX = hypotenuse * Math.cos(argument);
+		double legY = hypotenuse * Math.sin(argument);
+		// returning the result as a point
+		return new Point(legX, legY);
+	}
+
```

Comment:
plz change it back. if you want to, add an explanation about polar and cartesian presentations, but its really hard coded to vectors and sh*t, so show it. also, the name makes me think about 2017 autonomous command groups

Reply:
fixed